### ISSUE TITLE
Release 5.0.0-beta001

### ISSUE THREAD

**Issue by ghost** [castleproject/Windsor#418]
Can we release v5? 

Any outstanding issues we need to look at? 

/cc @stakx, @jonorossi

**Comment by fir3pho3nixx** [comment]
Sorry @stak accidentally tagged the wrong guy! Commuting and doing github on a mobile is apparently challenging - LOL

**Comment by fir3pho3nixx** [comment]
I think the MVC facility is not publishing it's symbols. :/

**Comment by jonorossi** [comment]
@fir3pho3nixx #346 is still assigned to the milestone in GitHub, happy for the milestone to be unassigned.

Did you want to release an official "5.0.0-beta1" to nuget.org to make sure everything with packaging works well giving you an easy chance to fix before final?

I don't think there is anything else I was expecting to be in 5.0, there is always more that can be done but there is no need to hold the ASP.NET Core facility up on that.

**Comment by fir3pho3nixx** [comment]
Forgot about that one! Do you mind if I have a quick refactor just to see how big this one is, if it is tiny we should definitely try and see if we can squeeze this one in. Let me get back to you.

**Comment by fir3pho3nixx** [comment]
After re-reading the issue I remember now.

We were going to solve #346 at the surface level API's using a "Cheshire the cat" pattern. Let me put a PR together with dotMemory profiler results because we agreed to using a "copy on write" strategy here. That will hopefully inform us as to whether we can include it in this release or whether we need to do more work achieve this. 

V5 on hold PR.

**Comment by fir3pho3nixx** [comment]
> Did you want to release an official "5.0.0-beta1" to nuget.org to make sure everything with packaging works well giving you an easy chance to fix before final?

Yes, please. 

It also gives me a chance to start drafting the solution for https://github.com/danielpalme/IocPerformance/issues/81. 

I would also like to delete the copy of this facility in our code base at work. Hoping our [New Relic](https://newrelic.com/) monitoring will give us some last chance saloon for clearing things up.

**Comment by fir3pho3nixx** [comment]
@generik0, @xhafan If you can also help me validate the beta I would appreciate it heaps. 

Thanks for all your help so far.

**Comment by generik0** [comment]
@fir3pho3nixx i Can. I will intergrate tomorrow and let it burn :-)

**Comment by xhafan** [comment]
@fir3pho3nixx, I'm on it ;)

**Comment by xhafan** [comment]
Works like a charm. I tested it with the latest code from https://github.com/castleproject/Windsor, master branch (commit [81fbdf084d3b6e235ff54c4f29bbc8d93fd408ec](https://github.com/castleproject/Windsor/commit/81fbdf084d3b6e235ff54c4f29bbc8d93fd408ec)).

In my app, the [middleware is registered](https://github.com/xhafan/emailmaker/blob/master/src/EmailMaker.WebsiteCore/Startup.cs#L96) like this:
```c#
_windsorContainer.Register(
                Component.For<IUnitOfWorkFactory>().AsFactory(),
                Component.For<TransactionScopeUnitOfWorkMiddleware>()
                        .DependsOn(Dependency.OnValue<System.Transactions.IsolationLevel>(System.Transactions.IsolationLevel.Serializable))
                        .LifestyleSingleton().AsMiddleware()
);
```
Windsor then creates an instance of [TransactionScopeUnitOfWorkMiddleware](https://github.com/xhafan/emailmaker/blob/master/src/EmailMaker.WebsiteCore/Middleware/TransactionScopeUnitOfWorkMiddleware.cs) with [IUnitOfWorkFactory](https://github.com/xhafan/emailmaker/blob/master/src/EmailMaker.WebsiteCore/Middleware/IUnitOfWorkFactory.cs) injected into it.

**Comment by MAliM1988** [comment]
Hi 

What time is the release date of the new version?

**Comment by fir3pho3nixx** [comment]
@MAliM1988 

We are investigating https://github.com/castleproject/Windsor/pull/420. Once that is done, I don't see anything holding us back.

Thanks for the interest :)

**Comment by fir3pho3nixx** [comment]
State of play:

1. API UX fail for [WindsorRegistrationOptions](https://github.com/castleproject/Windsor/blob/master/src/Castle.Facilities.AspNetCore/WindsorRegistrationOptions.cs#L25) in #423. Should be fixed before beta.
2. SomeFuckedUpShitInWindsorIWantToFix in #420. Not sure what we do with this can we see what happens with typed factories first. Will update the docs for #420.

**Comment by fir3pho3nixx** [comment]
State of play:

1. API UX fail for [WindsorRegistrationOptions](https://github.com/castleproject/Windsor/blob/master/src/Castle.Facilities.AspNetCore/WindsorRegistrationOptions.cs#L25) in #423. @MAliM1988 closed the issue so I am adding this to aspnet vNext here(https://github.com/fir3pho3nixx/Windsor/commit/14ad698c7df4eec8fa7c231807a622467e2bec11).
2. After [this](https://github.com/castleproject/Windsor/pull/420#issuecomment-409755168) We are mulling over deprecating Resolve(object) because it is too general and causing problems with immutability.

**Comment by fir3pho3nixx** [comment]
State of play:

1. API UX fail for [WindsorRegistrationOptions](https://github.com/castleproject/Windsor/blob/master/src/Castle.Facilities.AspNetCore/WindsorRegistrationOptions.cs#L25) in #423. @MAliM1988 re-opened! Awaiting PR as per comment [here](https://github.com/castleproject/Windsor/issues/423#issuecomment-413641184).

2. Final commits are in for #420 and awaiting review, we have deprecated object/IDictionary as parameters for IWindsorContainer.Resolve in favour of using Arguments with static factory methods instead(https://github.com/fir3pho3nixx/Windsor/blob/deprecate-resolve-with-non-generic-dictionary/docs/arguments.md).

**Comment by fir3pho3nixx** [comment]
State of play:

1. API UX fail for [WindsorRegistrationOptions](https://github.com/castleproject/Windsor/blob/master/src/Castle.Facilities.AspNetCore/WindsorRegistrationOptions.cs#L25) in #423. A PR has been submitted to fix this #432.

2. Commits for #420 and have been reviewed, and need some changes from myself.

**Comment by fir3pho3nixx** [comment]
Final commits are in for #420. Just needs squishing. 

Also done some more testing with disposables. 

I think we are good to release now.

**Comment by fir3pho3nixx** [comment]
@jonorossi raised some issues that need addressing on #420. I also do not think the documentation is up to scratch either. I think we should release ASP.NET Core facility in the meantime. This simply needs more time.

@jonorossi do think we can do a pre-release?

**Comment by beginor** [comment]
waiting for windsor 5.0.0 release！

**Comment by tapmantwo** [comment]
Hi, any news on a release? I'm keen to try the new aspnet core facility.

**Comment by stakx** [comment]
@tapmantwo, there's still one pending PR (#440) which needs to be merged first; quoting from https://github.com/castleproject/Windsor/pull/444#issue-222732311:

> [...] I'll get this merged and we can push out a beta for 5.0 with the ASP.NET Core changes.

**Comment by jonorossi** [comment]
#444 is now merged.

@castleproject/committers (just you @stakx now) is everyone happy with a `5.0.0-beta001` release?

**Comment by stakx** [comment]
@jonorossi - (with my best Super Mario Brothers Luigi voice:) let's-a-go!

**Comment by jonorossi** [comment]
Ran into a problem with our build scripts, I guess we haven't released a beta in a while. Looking into it now.

> obj\Release\netstandard1.6\Castle.Windsor.AssemblyInfo.cs(15,59): error CS7035: The specified version string does not conform to the recommended format - major.minor.build.revision [C:\projects\windsor\src\Castle.Windsor\Castle.Windsor.csproj]
> obj\Release\net45\Castle.Windsor.AssemblyInfo.cs(15,59): error CS7035: The specified version string does not conform to the recommended format - major.minor.build.revision [C:\projects\windsor\src\Castle.Windsor\Castle.Windsor.csproj]

**Comment by jonorossi** [comment]
- Tagged [`v5.0.0-beta001`](https://github.com/castleproject/Windsor/releases/tag/v5.0.0-beta001)
- Confirmed successfully pushed to nuget.org, except `Castle.Facilities.AspNetCore` which unfortunately someone ("buddalasunil") has published a fork and unlisted it with the version "4.1.0". I've emailed them to get ownership transferred, will keep everyone informed on progress. (https://www.nuget.org/packages/Castle.Facilities.AspNetCore/4.1.0)
- Tweeted (will do once the last package is up)

**Comment by jonorossi** [comment]
- We've now got ownership of [Castle.Facilities.AspNetCore](https://www.nuget.org/packages/Castle.Facilities.AspNetCore/) and it has been pushed.
- [Tweeted](https://twitter.com/castleproject/status/1056001405591314432)

Renamed this issue since it isn't the final 5.0.0 release. We'll create a new issue for that later once people have had a chance to play with the bits. Please get involved if you have problems or see areas for improvements.

Thanks again to everyone that contributed to this release.

**Comment by tapmantwo** [comment]
Fantastic, thank you.
